### PR TITLE
refactor(preventDoubleAction): update anonymous component feature introduced in circom 2.1.0

### PR DESCRIPTION
**Pull request by moven0831**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`yarn build`) was run locally in root directory and any changes were pushed
- [ ] Lint (`yarn lint:fix`) was run locally in root directory and any changes were pushed

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

This PR modifies the circom styling from previous version to the newest (^2.1.0).

Previous Version
```
template A(n){
   signal input a, b;
   signal output c;
   c <== a*b;
}
template B(n){
   signal input in[n];
   signal out;
   component temp_a = A(n);
   temp_a.a <== in[0]; 
   temp_a.b <== in[1];
   out <== temp_a.c;
}
component main = B(2);
```

New version
```
template A(n){
   signal input a, b;
   signal output c;
   c <== a*b;
}
template B(n){
   signal input in[n];
   signal out <== A(n)(in[0],in[1]);
}
component main = B(2);
```

By doing so, we could increase the readability of the code and make the code much neater to maintain.

Issue Number: #530 

## What is the new behavior?

The behavior of the modification should not change any result from the previous version.  

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by Yingyu18 — APPROVED** [review]
LGTM.
Passed the test.

File: packages/circuits/circuits/preventDoubleAction.circom [hunk 1]
```diff
@@ -33,32 +33,22 @@ template PreventDoubleAction(STATE_TREE_DEPTH, EPOCH_KEY_NONCE_PER_EPOCH, FIELD_
...
-    poseidon.inputs[0] <== identity_nullifier;
-    poseidon.inputs[1] <== external_nullifier;
-    nullifier <== poseidon.out;
+    nullifier <== Poseidon(2)([identity_nullifier, external_nullifier]);
 
      /* 2. Compute identity commitment */
-    component commitment = IdentityCommitment();
-    commitment.nullifier <== identity_nullifier;
-    commitment.trapdoor <== identity_trapdoor;
+    signal commitment_secret;
```
Comment (FoodChain1028) [inline packages/circuits/circuits/preventDoubleAction.circom:39]: I suggest using `identity_secret` to config the same signal in `IdentityCommitment()`

File: packages/circuits/package.json [hunk 2]
```diff
@@ -29,7 +29,8 @@
         "buildTs": "tsc",
         "copyCircom": "rm -rf ./dist/zksnarkBuild && cp -rf ./zksnarkBuild ./dist/zksnarkBuild",
         "time": "ts-node scripts/time.ts",
-        "test": "mocha -r ts-node/register test/*.test.ts --exit"
+        "test": "mocha -r ts-node/register test/*.test.ts --exit",
+        "refresh": "find ./zksnarkBuild -type f -not -name 'powersOfTau28_hez_final_18.ptau' -delete && yarn build"
```
Comment (FoodChain1028) [inline packages/circuits/package.json:33]: it can be modified into the newest version!

**Review by FoodChain1028 — CHANGES_REQUESTED** [review]
Looks Great!! Yet there are several parts that can be improved in my view 😆.

**Review by C1em3nt — APPROVED** [review]
LGTM

File: packages/circuits/circuits/preventDoubleAction.circom (hunk 1 above)
Comment (moven0831) [reply packages/circuits/circuits/preventDoubleAction.circom]: That makes sense. Thanks for review

**Review by moven0831 — COMMENTED** [review]
Hi @FoodChain1028 , I've renamed variable `commitment_secret` to `identity_secret`.

**Review by FoodChain1028 — APPROVED** [review]
great job!

**Pull request merged**